gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/authentication/filter/JWTAuthenticationFilter.java

```diff
@@ -102,26 +103,26 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
                         .findAny();
             }
 
-            if (!optionalStringToken.isPresent() || !optionalStringToken.get().getValue().startsWith("Bearer ")) {
+            if (optionalStringToken.isEmpty() || !optionalStringToken.get().getValue().startsWith("Bearer ")) {
                 throw new BadCredentialsException("No JWT token found");
             }
             authToken = optionalStringToken.get().getValue().substring(7);
         }
 
         try {
-            JWT payload = jwtParser.parse(authToken);;
+            JWT payload = jwtParser.parse(authToken);
             Map<String, Object> claims = new HashMap<>(payload);
             claims.put(Claims.ip_address, remoteAddress(request));
             claims.put(Claims.user_agent, userAgent(request));
+
             DefaultUser user = new DefaultUser((String) claims.get(StandardClaims.PREFERRED_USERNAME));
             user.setId((String) claims.get(StandardClaims.SUB));
-            user.setRoles(user.getRoles() != null ? user.getRoles() : (List<String>) claims.get(CustomClaims.ROLES));
```

Comment:
Are you sure that the roles are not used later in AuthenticationServiceImpl.onAuthenticationSuccess ?

Reply:
FP